### PR TITLE
webgpu: export webgpu utilities

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6707)

### CODE REVIEW

**Comment by xhcao** [comment]
@haoyunfeix Please help to review it, thank you.

File: tfjs-backend-webgpu/src/index.ts [hunk 1]
```diff
@@ -49,11 +49,12 @@ if (isWebGPUSupported()) {
...
-      deviceDescriptor.requiredFeatures = ['timestamp-query' as const];
+      deviceDescriptor.requiredFeatures = ['timestamp-query' as const ];
     }
     const device: GPUDevice = await adapter.requestDevice(deviceDescriptor);
     return new WebGPUBackend(device);
   }, 3 /*priority*/);
 }
 
 export {webgpu};
+export * from './base';
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/index.ts:60]: why not directly call `export * from './webgpu';` here?

Comment (xhcao) [reply tfjs-backend-webgpu/src/index.ts:60]: Refer it to webgl backend, we could put all wanted export utilities in base.ts file in future.

**Comment by xhcao** [comment]
Hi, @qjia7 I do not know the reason why here uses the `base.ts` for webgl, why not import and export directly in index.ts.   @pyu10055 @lina128 Could you help to answer the question? thank you.

File: tfjs-backend-webgpu/src/index.ts [hunk 2]
```diff
@@ -49,11 +49,12 @@ if (isWebGPUSupported()) {
     };
 
     if (supportTimeQuery) {
-      deviceDescriptor.requiredFeatures = ['timestamp-query' as const];
+      deviceDescriptor.requiredFeatures = ['timestamp-query' as const ];
```
[the remark this answers is not in the document]
Comment (xhcao) [reply tfjs-backend-webgpu/src/index.ts:52]: I had noticed this extra space, it was added automatically by vs editor.

File: tfjs-backend-webgpu/src/index.ts (hunk 1 above)
Comment (mattsoulanille) [reply tfjs-backend-webgpu/src/index.ts:60]: The webgl backend puts kernels into the `register_all_kernels.ts` file and everything else into `base.ts`. This is for better readability, and makes sure any side effects from setting up the backend happen before kernels are registered (I'm not 100% sure this is necessary). If you want to do the same for webgpu, you could move everything from index.ts into base.ts. However, it's a style choice, and if you prefer to keep everything in index.ts, I think that's fine.

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

**Review by gyagp — APPROVED** [review]
LGTM

**Pull request merged**